src/ClassFinder.php

```diff
@@ -45,6 +45,10 @@ public function getDeclaredClasses() : Collection
             return starts_with($value, 'Illuminate'); // Ignore Illuminate Packages
         })->reject(function ($value, $key) {
             return starts_with($value, 'Symfony');
+        })->reject(function ($class) {
+            return (new ReflectionClass($class))->isNative();
+        })->reject(function ($class) {
+            return (new ReflectionClass($class))->isVendorProvided($this->basePath);
```

Comment:
I would actually like to map these to ReflectionClass instances up front and transform these checks to operate on that.

I only did this to change as little as possible. Would that be a worthwhile refactoring for this PR or should it be separate? (If the checks do end up living here at the end)

Reply:
I think I should move those checks to an entire different class. This class is just responsible for finding the classes. I think I will add some sort of `RejectionFilter`-class (*naming is hard*) which will remove all classes which should be ignored.

I originally had the idea to allow the user to configure a set of Namespaces which be ignored. I currently have hardcoded `Illuminate` and `Symfony`, but I would like to make that configurable; but this would be too much for this PR. I will prepare a refactor PR in the upcoming days.

Reply:
Yeah, that sounds reasonable.